File: AIO/AIO 2025/[p1] Soccer Match.cpp
#include <bits/stdc++.h>
using namespace std;
 
int main() {    
    int N;
    cin >> N;
    int goal;
    output = "NO";
    int team1 = 0;
    int team2 = 0;
 
    for(int i = 0; i < N; i++) {
        cin >> goal;
        if(goal == 1) team1++;
        else team2++;
      
        if(team1 > team2) output = "YES";
    }
 
    cout << output;
}

File: AIO/AIO 2025/[p1] Soccer Match.py
N = int(input())
goals = list(map(int, input().split()))
 
output = False
 
# Run stimulation
team1 = 0
team2 = 0
 
for i in goals:
    if i == 1:
        team1 += 1
    else:
        team2 += 1
    if team1 > team2:
        output = True
 
# Output
if output:
    print('YES')
else:
    print('NO')
